common/repositories/inventory_snapshots_repository.h

```diff
@@ -68,6 +95,227 @@ class InventorySnapshotsRepository: public BaseInventorySnapshotsRepository {
 
 		return count;
 	}
+
+	static void ClearCharacterInvSnapshots(Database &db, uint32 character_id, bool from_now)
+	{
+		uint32 del_time = time(nullptr);
+		if (!from_now) {
+			del_time -= RuleI(Character, InvSnapshotHistoryD) * 86400;
+		}
+
+		DeleteWhere(db, fmt::format("`character_id` = '{}' AND `time_index` <= '{}'", character_id, del_time));
+	}
+
+	static void ListCharacterInvSnapshots(Database &db, uint32 character_id, std::list<std::pair<uint32, int>> &is_list)
+	{
+		const std::string &query = fmt::format(
+			"SELECT `time_index`, COUNT(*) FROM `inventory_snapshots` WHERE "
```

Comment:
Numbers shouldn't be quoted in queries, it throws off the query planner

Reply:
Updated